src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -417,6 +397,34 @@ public int Seek(int char0, int char1, int char2)
             }
         }
 
+        private static int FindFirstEqualByte(Vector<byte> chEquals)
+        {
+            // Quasi-tree search
+            var vector64 = Vector.AsVectorInt64(chEquals);
+            for (var i = 0; i < Vector<long>.Count; i++)
+            {
+                var longValue = vector64[i];
+                if (longValue == 0) continue;
+
+                var shift = i << 1;
+                var offset = shift << 2;
+                var vector32 = Vector.AsVectorInt32(chEquals);
```

Comment:
I did the microbenchmarking myself ([Program.cs](https://gist.github.com/halter73/236f9f300b7f5cfe1d40)), and it appears that it's better to do the "Quasi-tree search". At least it doesn't seem any worse. Here are the results from my laptop:

    C:\Users\shalter\dev\Universe\KestrelHttpServer\samples\ConsoleApp1 [dev +1 ~2 -0 !]> # FindFirstEqualByte
    C:\Users\shalter\dev\Universe\KestrelHttpServer\samples\ConsoleApp1 [dev +1 ~2 -0 !]> dnx --configuration Release ConsoleApp1
    1635
    C:\Users\shalter\dev\Universe\KestrelHttpServer\samples\ConsoleApp1 [dev +1 ~2 -0 !]> dnx --configuration Release ConsoleApp1
    1640
    C:\Users\shalter\dev\Universe\KestrelHttpServer\samples\ConsoleApp1 [dev +1 ~2 -0 !]> dnx --configuration Release ConsoleApp1
    1453

    C:\Users\shalter\dev\Universe\KestrelHttpServer\samples\ConsoleApp1 [dev +1 ~2 -0 !]> # FindFirstEqualByte2
    C:\Users\shalter\dev\Universe\KestrelHttpServer\samples\ConsoleApp1 [dev +1 ~2 -0 !]> dnx --configuration Release ConsoleApp1
    1715
    C:\Users\shalter\dev\Universe\KestrelHttpServer\samples\ConsoleApp1 [dev +1 ~2 -0 !]> dnx --configuration Release ConsoleApp1
    1697
    C:\Users\shalter\dev\Universe\KestrelHttpServer\samples\ConsoleApp1 [dev +1 ~2 -0 !]> dnx --configuration Release ConsoleApp1
    1688

Merging now.